network/ethernet_frame.ksy

```diff
@@ -48,6 +44,20 @@ instances:
       first location bears special marker (0x8100), then it is not the
       real ether frame yet, an additional payload (`tci`) is expected
       and real ether type is upcoming next.
+  body:
+    io: _root._io
+    pos: _root._io.size - body_and_checksum.length
+    size: body_and_checksum.length - 4
+    type:
+      switch-on: ether_type
+      cases:
+        'ether_type_enum::ipv4': ipv4_packet
+        'ether_type_enum::ipv6': ipv6_packet
+  checksum:
```

Comment:
What is the checksum algorithm? If it's CRC-32, I'd prefer calling the field `crc32` (instead of ~~`checksum`~~).

Also, it would be great if you can provide a parse instance for the input byte array of the checksum, preferably called "name of the checksum field + `_input`" (e.g. `crc32_input`). This will make it much easier to check if the checksum is valid (in the application code, or even manually in a hex viewer).

Reply:
> What is the checksum algorithm? If it's CRC-32, I'd prefer calling the field crc32 (instead of checksum).

I wanted to use the same naming as used by TCP or UDP. But I am also fine with `crc32` instead of `checksum`.

> Also, it would be great if you can provide a parse instance for the input byte array of the checksum, preferably called "name of the checksum field + _input" (e.g. crc32_input). This will make it much easier to check if the checksum is valid (in the application code, or even manually in a hex viewer).

I am not sure what you mean with "parse instance". An example bytearray to test the code?